### ISSUE TITLE
Explodes without commit SHA1

### ISSUE THREAD

**Issue by AlekSi** [drone-plugins/drone-slack#28]
Drone 0.5.0+dev, running `drone exec` without `--commit.sha=` explodes:

```
[slack:L0:0s] panic: runtime error: slice bounds out of range [recovered]
[slack:L1:0s]   panic: runtime error: slice bounds out of range
[slack:L2:0s]
[slack:L3:0s] goroutine 1 [running]:
[slack:L4:0s] panic(0x81fc60, 0xc82000a050)
[slack:L5:0s]   /usr/local/go/src/runtime/panic.go:481 +0x3e6
[slack:L6:0s] github.com/drone-plugins/drone-slack/vendor/github.com/urfave/cli.HandleAction.func1(0xc8200d15e8)
[slack:L7:0s]   /go/src/github.com/drone-plugins/drone-slack/vendor/github.com/urfave/cli/app.go:476 +0x417
[slack:L8:0s] panic(0x81fc60, 0xc82000a050)
[slack:L9:0s]   /usr/local/go/src/runtime/panic.go:443 +0x4e9
[slack:L10:0s] main.message(0x0, 0x0, 0x0, 0x0, 0xc82000bf64, 0x4, 0x0, 0x0, 0x0, 0xc82000bfb0, ...)
[slack:L11:0s]  /go/src/github.com/drone-plugins/drone-slack/plugin.go:83 +0x407
[slack:L12:0s] main.Plugin.Exec(0x0, 0x0, 0x0, 0x0, 0xc82000bf64, 0x4, 0x0, 0x0, 0x0, 0xc82000bfb0, ...)
[slack:L13:0s]  /go/src/github.com/drone-plugins/drone-slack/plugin.go:46 +0x7d
[slack:L14:0s] main.run(0xc8200c4f00, 0x0, 0x0)
[slack:L15:0s]  /go/src/github.com/drone-plugins/drone-slack/main.go:152 +0x79c
[slack:L16:0s] reflect.Value.call(0x77e680, 0x96daa8, 0x13, 0x8a69f8, 0x4, 0xc8200d1548, 0x1, 0x1, 0x0, 0x0, ...)
[slack:L17:0s]  /usr/local/go/src/reflect/value.go:435 +0x120d
[slack:L18:0s] reflect.Value.Call(0x77e680, 0x96daa8, 0x13, 0xc8200d1548, 0x1, 0x1, 0x0, 0x0, 0x0)
[slack:L19:0s]  /usr/local/go/src/reflect/value.go:303 +0xb1
[slack:L20:0s] github.com/drone-plugins/drone-slack/vendor/github.com/urfave/cli.HandleAction(0x77e680, 0x96daa8, 0xc8200c4f00, 0x0, 0x0)
[slack:L21:0s]  /go/src/github.com/drone-plugins/drone-slack/vendor/github.com/urfave/cli/app.go:485 +0x2ee
[slack:L22:0s] github.com/drone-plugins/drone-slack/vendor/github.com/urfave/cli.(*App).Run(0xc820001080, 0xc82000a200, 0x1, 0x1, 0x0, 0x0)
[slack:L23:0s]  /go/src/github.com/drone-plugins/drone-slack/vendor/github.com/urfave/cli/app.go:244 +0xb05
[slack:L24:0s] main.main()
[slack:L25:0s]  /go/src/github.com/drone-plugins/drone-slack/main.go:116 +0x11d5
```

It works with `--commit.sha=…`. Probably error message should be better.

**Comment by vtolstov** [comment]
@bradrydzewski this plugin dies when drone triggered by tag push, because DRONE_COMMIT_SHA does not exposed

**Comment by bradrydzewski** [comment]
@vtolstov pull requests welcome

**Comment by appleboy** [comment]
https://github.com/drone-plugins/drone-slack/pull/50 resolve this issue.

**Comment by tboerger** [comment]
Solved via #50
